File: Test_3.Task_3/Test_3.Task_3/ComplexNumber.cs
namespace Test_3.Task_3

{
  /// <summary>
  /// This class created for work with complex number.
  /// </summary>
  public class ComplexNumber
  {
    private double realPart;
    private double imaginaryPart;

    /// <summary>
    ///This constructor initializes complex number (0, 0).
    /// </summary>
    public ComplexNumber()
    {
      realPart = 0.0;
      imaginaryPart = 0.0;
    }

    /// <summary>
    /// This constructor initializes the parts of complex number by parameter values.
    /// </summary>
    /// <param name="realPart">Real part of complex number.</param>
    /// <param name="imaginaryPart">Imaginary part of complex number.</param>
    public ComplexNumber(double realPart, double imaginaryPart)
    {
      this.realPart = realPart;
      this.imaginaryPart = imaginaryPart;
    }

    /// <summary>
    /// This constructor copies value of created object in new object.
    /// </summary>
    /// <param name="complexNumber">Created object.</param>
    public ComplexNumber(ComplexNumber complexNumber)
    {
      realPart = complexNumber.realPart;
      imaginaryPart = complexNumber.imaginaryPart;
    }

    public double RealPart { get { return realPart; } set { realPart = value; } }
    public double ImaginaryPart { get { return imaginaryPart; } set { imaginaryPart = value; } }

    /// <summary>
    /// This method sums two complex numbers.
    /// </summary>
    /// <param name="firstNumber">First complex number.</param>
    /// <param name="secondNumber">Second complex number.</param>
    /// <returns>Returns sum of two complex numbers.</returns>
    public ComplexNumber FoldComplexNumbers(ComplexNumber firstNumber, ComplexNumber secondNumber)
    {
      ComplexNumber result = new ComplexNumber();
      result.realPart = firstNumber.realPart + secondNumber.realPart;
      result.imaginaryPart = firstNumber.imaginaryPart + secondNumber.imaginaryPart;
      return result;
    }

    /// <summary>
    /// This method multiplies two complex numbers.
    /// </summary>
    /// <param name="firstNumber">First complex number.</param>
    /// <param name="secondNumber">Second complex number.</param>
    /// <returns>Returns multiply of two complex numbers.</returns>
    public ComplexNumber MultiplyComplexNumbers(ComplexNumber firstNumber, ComplexNumber secondNumber)
    {
      ComplexNumber result = new ComplexNumber();
      result.realPart = firstNumber.realPart * secondNumber.realPart - firstNumber.imaginaryPart * secondNumber.imaginaryPart;
      result.imaginaryPart = firstNumber.imaginaryPart * secondNumber.realPart + firstNumber.realPart * secondNumber.ImaginaryPart;
      return result;
    }
  }
}

File: Test28.11/src/ComplexNumber.java
/**
 * This class is working with complex number.
 */
public class ComplexNumber {
  private double realPart;
  private double imaginaryPart;

  /**
   * This constructor initializes the parts of the complex number by (0.0).
   */
  public ComplexNumber() {
    realPart = 0.0;
    imaginaryPart = 0.0;
  }

  /**
   * This constructor initializes the parts of the complex number by parameters values.
   *
   * @param realPart      the real part of the complex number
   * @param imaginaryPart the imaginary part of the complex number
   */
  public ComplexNumber(double realPart, double imaginaryPart) {
    this.realPart = realPart;
    this.imaginaryPart = imaginaryPart;
  }

  /**
   * This constructor copies the value of the created object to the newly created object.
   *
   * @param complexNumber the created object
   */
  public ComplexNumber(ComplexNumber complexNumber) {
    this(complexNumber.getRealPart(), complexNumber.getImaginaryPart());
  }

  public double getRealPart() {
    return realPart;
  }

  public void setRealPart(double realPart) {
    this.realPart = realPart;
  }

  public double getImaginaryPart() {
    return imaginaryPart;
  }

  public void setImaginaryPart(double imaginaryPart) {
    this.imaginaryPart = imaginaryPart;
  }

  /**
   * Sums up two complex numbers.
   *
   * @param firstNumber  the first complex number
   * @param secondNumber the second complex number
   * @return summation result
   */
  public ComplexNumber addition(ComplexNumber firstNumber, ComplexNumber secondNumber) {
    return new ComplexNumber(firstNumber.realPart + secondNumber.realPart, firstNumber.imaginaryPart + secondNumber.imaginaryPart);
  }

  /**
   * Subtracts two complex numbers.
   *
   * @param firstNumber  the first complex number
   * @param secondNumber the second complex number
   * @return subtraction result
   */
  public ComplexNumber subtraction(ComplexNumber firstNumber, ComplexNumber secondNumber) {
    return new ComplexNumber(firstNumber.realPart - secondNumber.realPart, firstNumber.imaginaryPart - secondNumber.imaginaryPart);
  }

  /**
   * Multiplies two complex numbers.
   *
   * @param firstNumber  the first complex number
   * @param secondNumber the second complex number
   * @return multiplication result
   */
  public ComplexNumber multiplication(ComplexNumber firstNumber, ComplexNumber secondNumber) {
    ComplexNumber result = new ComplexNumber();
    result.realPart = firstNumber.realPart * secondNumber.realPart - firstNumber.imaginaryPart * secondNumber.imaginaryPart;
    result.imaginaryPart = firstNumber.realPart * secondNumber.imaginaryPart + firstNumber.imaginaryPart * secondNumber.realPart;
    return result;
  }

  /**
   * Divides two complex numbers.
   *
   * @param firstNumber  the first complex number
   * @param secondNumber the second complex number
   * @return result of division
   */
  public ComplexNumber division(ComplexNumber firstNumber, ComplexNumber secondNumber) {
    // to do - realization of division
    return new ComplexNumber();
  }

  private double moduleComplexNumber(ComplexNumber complexNumber) {
    return Math.sqrt(Math.pow(complexNumber.realPart, 2) + Math.pow(complexNumber.imaginaryPart, 2));
  }

  public int comparison(ComplexNumber firstNumber, ComplexNumber secondNumber) {
    return (Double.compare(moduleComplexNumber(firstNumber), moduleComplexNumber(secondNumber)));
  }
}
